cql3/statements/create_function_statement.cc

```diff
@@ -47,7 +47,7 @@ shared_ptr<functions::function> create_function_statement::create(query_processo
             std::move(return_type), _called_on_null_input, std::move(ctx));
     } else if (_language == "xwasm") {
        // FIXME: need better way to test wasm compilation without real_database()
-       wasm::context ctx{db.real_database().wasm_engine(), _name.name};
+       wasm::context ctx{db.real_database().wasm_engine(), _name.name, db.real_database().udf_cache()};
```

Comment:
udf_cache shoudn't be in replica::database. That's a replica-side thing, while functions can be executed in any role (but mostly coordinator and super-coordinator).

Reply:
We need to have it reachable from this position, what place would you suggest to put it in? `query_processor`? Maybe even `storage_proxy`. Also, should we remove the wasm_engine from the replica::database while we're at it?

Reply:
I moved it to the `query_processor`. Because the wasm_engine isn't strictly connected to the instance caching, let's make it a follow-up (it will also depend on the addition of rust wasmtime)